### DISCUSSION TITLE
Live Update: Setup in React?

### DISCUSSION THREAD

**Discussion by hugocxl** [capawesome-team/capacitor-plugins#318] [Q&A] [answered]
Hi there, 👋!
I'm trying to implement the Live Update plugin in a Capacitor+React app, but I cannot make it work after carefully following all the instructions in the docs.

I'm getting always this error: `[ LiveUpdate ]  App is not ready. Default bundle is already in use.`

I created this hook and I'm calling it in the root node of the app to handle the sync.

```tsx
import { useEffect } from 'react'
import { LiveUpdate } from '@capawesome/capacitor-live-update'

export const useAutoUpdater = () => {
	useEffect(() => {
		window.alert('Auto updater')

		const initialize = async () => {
			try {
				// Mark the app as ready on mount
				window.alert('Marking app as ready')
				await LiveUpdate.ready()
				window.alert('App marked as ready')

				// Check for updates
				const result = await LiveUpdate.sync()
				if (result.nextBundleId) {
					window.alert('Update available')
					await LiveUpdate.reload()
					window.alert('Update applied')
				}
			} catch (error) {
				console.error('Auto update failed:', error)
			}
		}

		initialize()
	}, [])
}
```

Am I missing something? Is there any working React example?

Thanks for the help and this amazing plugin ❤️

**Comment by robingenz** [comment]
Hi @hugocxl, thank you for creating this discussion! Sadly, I have absolutely no experience with React. The `ready()` method must be called as early as possible to prevent a reset of the bundle. This reset is performed for backup purposes so that the app continues to work even if the developer has deployed an invalid bundle. By default, you have 10 seconds to do this (see `readyTimeout` configuration option). So it seems that your code is not executed within this time.

**Comment by hugocxl** [comment, marked as answer]
Hi @robingenz! Thanks for your quick answer ❤️.
My bad here 🤦🏻‍♂️, I generated a first bundle in the cloud service that did not included the `ready` call, therefore never being able to replace that app (even though I was changing the code afterwards).

Working great now! 

Thanks again for this great lib and sorry for the inconveniences 🙏

**Reply by robingenz** [reply]
You're welcome!
